cocogen/ccngen/ccn_copy.c

```diff
@@ -10,12 +10,19 @@
 #define MAYBE_UNUSED
 #endif
 
+#ifdef __GNUC__
```

Comment:
I've looked into the maybe unused not working further, and I am not able to reproduce it.

Using both GCC 11.4 and GCC 13.2, I've tested running:
```bash
git clone git@github.com:CoCoNut-UvA/civicc-skeleton.git
cd civicc-skeleton
./configure.sh
make -C build-debug
```

I don't get any warnings from the user_warn variable being unused.

I've also created an minimal example on godbolt here that shows it works back to GCC 6.1: [https://godbolt.org/z/hb8xY1Tsd](https://godbolt.org/z/hb8xY1Tsd).

Could you please run the above instructions, see if you get any warnings, and if so post the GCC version you are using?

I would rather keep using MAYBE_UNUSED, so we can easily switch to [C23 [[maybe_unused]]](https://en.cppreference.com/w/c/language/attributes/maybe_unused) once C23 will become commonplace in the major compilers.